src/Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -320,7 +342,23 @@ internal partial class TokenAcquisition : ITokenAcquisitionInternal
                 }
             }
 
-            return builder.ExecuteAsync();
+            try
+            {
+                return builder.ExecuteAsync();
+            }
+            catch (MsalServiceException exMsal) when (IsInvalidClientCertificateError(exMsal))
+            {
+                DefaultCertificateLoader.ResetCertificates(_microsoftIdentityOptions.ClientCertificates);
+                _application = null;
+
+                // Retry
+                retryClientCertificate = true;
+                return GetAuthenticationResultForAppAsync(scope, tenant, tokenAcquisitionOptions);
```

Comment:
>GetAuthenticationResultForAppAsync [](http://example.com/codeflow?start=23&length=34)

we have a different way of doing the logs now...see line 629 (another one i can fix up)